uint8[3]     robot_roles
uint8        game_state
